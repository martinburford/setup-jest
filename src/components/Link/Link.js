import React, { Component } from 'react';

const STATUS = {
  ERROR: 'error',
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

class Link extends Component {
  constructor() {
    super();

    this._onClick = this._onClick.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onClick(){
    this.setState({class: STATUS.HOVERED});
  }

  _onMouseLeave(){
    this.setState({class: STATUS.NORMAL});
  }

  componentDidMount(){
    if(this.props.error){
      this.setState({class: STATUS.ERROR});
    }
  }

  render(){
    return (
      <div>
        <a
          className={this.state.class}
          href={this.props.page || '#'}
          onClick={this._onClick}
          onMouseLeave={this._onMouseLeave}
        >
          {this.props.children}
        </a>
      </div>
    );
  }
}

export default Link;