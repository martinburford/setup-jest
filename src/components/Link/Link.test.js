import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';

it('<Link /> renders correctly (with no additional props)', () => {
  const tree = renderer
    .create(<Link page="http://www.site.com">Site</Link>)
    .toJSON();

  // Check the snapshot against the JSON tree
  expect(tree).toMatchSnapshot();
});

it('<Link /> renders correctly (with error prop)', () => {
  const tree = renderer
    .create(
      <Link page="http://www.site.com" error>Site</Link>
    )
    .toJSON();

  // Check the snapshot against the JSON tree
  expect(tree).toMatchSnapshot();
});