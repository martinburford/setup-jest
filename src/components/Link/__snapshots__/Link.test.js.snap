// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders correctly (with error prop) 1`] = `
<div>
  <a
    className="error"
    href="http://www.site.com"
    onClick={[Function]}
    onMouseLeave={[Function]}
  >
    Site
  </a>
</div>
`;

exports[`<Link /> renders correctly (with no additional props) 1`] = `
<div>
  <a
    className="normal"
    href="http://www.site.com"
    onClick={[Function]}
    onMouseLeave={[Function]}
  >
    Site
  </a>
</div>
`;
